### ISSUE TITLE
segmentation on test data where there is no ground truth.

### ISSUE THREAD

**Issue by ninjakx** [qubvel/segmentation_models.pytorch#103]
How to predict on test data where I don't have any ground truth because even using the dataset uses 
mask images.

```
test_dataset = Dataset(
    x_test_dir, 
    y_test_dir, 
    classes=CLASSES, 
    preprocessing=get_preprocessing(preprocess_input),
)

test_dataloader = Dataloder(test_dataset, batch_size=1, shuffle=False)

```

this function takes 2 inputs- image and its mask.
I tried to make the dataset only accept input image but then how to use this line preprocessing=get_preprocessing(preprocess_input)

I have basically two MRI dataset. One is for training and another for testing(where I have to crop the tumour part). Training is done. Model is also working but how to predict on test data.
I predicted like this:

```
for each in os.listdir('test/'):
    print(each)
    img = cv2.imread(os.path.join('test', each))
    img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
    img = preprocessing_fn(img)
    img = img.transpose(2, 0, 1)#.astype('float32')
    # img = img.astype('uint8')/255.0 <--- not using this.
    x = torch.from_numpy(np.expand_dims(img, 0))
    y = model.predict(x.to(DEVICE))
```

But the result I am getting is so bad.
![Screenshot from 2019-11-25 14-22-38](https://user-images.githubusercontent.com/29797787/69526173-a15a2180-0f8f-11ea-863b-faf6e4493255.png)
![Screenshot from 2019-11-25 14-22-16](https://user-images.githubusercontent.com/29797787/69526174-a15a2180-0f8f-11ea-9937-40532f5b7fa1.png)

**Comment by qubvel** [comment]
Hi, you can modify dataset to return only images
```python
class InferenceDataset(BaseDataset):
    """CamVid Dataset. Read images, apply augmentation and preprocessing transformations.
    
    Args:
        images_dir (str): path to images folder
        augmentation (albumentations.Compose): data transfromation pipeline 
            (e.g. flip, scale, etc.)
        preprocessing (albumentations.Compose): data preprocessing 
            (e.g. noralization, shape manipulation, etc.)
    """

    def __init__(
            self, 
            images_dir, 
            augmentation=None, 
            preprocessing=None,
    ):
        self.ids = os.listdir(images_dir)
        self.images_fps = [os.path.join(images_dir, image_id) for image_id in self.ids]

        self.augmentation = augmentation
        self.preprocessing = preprocessing
    
    def __getitem__(self, i):
        
        # read data
        image = cv2.imread(self.images_fps[i])
        image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
        
        # apply augmentations
        if self.augmentation:
            sample = self.augmentation(image=image)
            image = sample['image']
        
        # apply preprocessing
        if self.preprocessing:
            sample = self.preprocessing(image=image)
            image = sample['image']
            
        return image
        
    def __len__(self):
        return len(self.ids)
```

Than use something like that
```
model.eval()
with torch.no_grad():
    for images in dataset:
        masks = model(image.to(images))
```

**Comment by ChrisMasafu** [comment]
> Hi, you can modify dataset to return only images
> 
> ```python
> class InferenceDataset(BaseDataset):
>     """CamVid Dataset. Read images, apply augmentation and preprocessing transformations.
>     
>     Args:
>         images_dir (str): path to images folder
>         augmentation (albumentations.Compose): data transfromation pipeline 
>             (e.g. flip, scale, etc.)
>         preprocessing (albumentations.Compose): data preprocessing 
>             (e.g. noralization, shape manipulation, etc.)
>     """
> 
>     def __init__(
>             self, 
>             images_dir, 
>             augmentation=None, 
>             preprocessing=None,
>     ):
>         self.ids = os.listdir(images_dir)
>         self.images_fps = [os.path.join(images_dir, image_id) for image_id in self.ids]
> 
>         self.augmentation = augmentation
>         self.preprocessing = preprocessing
>     
>     def __getitem__(self, i):
>         
>         # read data
>         image = cv2.imread(self.images_fps[i])
>         image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
>         
>         # apply augmentations
>         if self.augmentation:
>             sample = self.augmentation(image=image)
>             image = sample['image']
>         
>         # apply preprocessing
>         if self.preprocessing:
>             sample = self.preprocessing(image=image)
>             image = sample['image']
>             
>         return image
>         
>     def __len__(self):
>         return len(self.ids)
> ```
> 
> Than use something like that
> 
> ```
> model.eval()
> with torch.no_grad():
>     for images in dataset:
>         masks = model(image.to(images))
> ```

Could you please clarify on this and put the whole code together for us newbies (predicting on our own image - which doesn't have a mask). Thank you professor.
